control/control.go

```diff
@@ -530,101 +505,136 @@ func (p *pluginControl) validatePluginSubscription(pl core.SubscribedPlugin) []s
 	return serrs
 }
 
-func (p *pluginControl) validateMetricTypeSubscription(mt core.RequestedMetric, cd *cdata.ConfigDataNode) []serror.SnapError {
-	var serrs []serror.SnapError
+// validateMetricTypeSubscription validates the subscription for given requested metric
+func (p *pluginControl) validateMetricTypeSubscription(rmt core.RequestedMetric, cdt *cdata.ConfigDataTree) []serror.SnapError {
+	serrs := []serror.SnapError{}
+
 	controlLogger.WithFields(log.Fields{
 		"_block":    "validate-metric-subscription",
-		"namespace": mt.Namespace(),
-		"version":   mt.Version(),
+		"namespace": rmt.Namespace(),
+		"version":   rmt.Version(),
 	}).Info("subscription called on metric")
 
-	m, err := p.metricCatalog.Get(mt.Namespace(), mt.Version())
+	// match the requested metric with the cataloged metrics types
+	mts, err := p.metricCatalog.GetMatchedMetricTypes(rmt.Namespace(), rmt.Version())
 
 	if err != nil {
 		serrs = append(serrs, serror.New(err, map[string]interface{}{
-			"name":    mt.Namespace().String(),
-			"version": mt.Version(),
+			"name":    rmt.Namespace().String(),
+			"version": rmt.Version(),
 		}))
 		return serrs
 	}
 
-	// No metric found return error.
-	if m == nil {
-		serrs = append(serrs, serror.New(fmt.Errorf("no metric found cannot subscribe: (%s) version(%d)", mt.Namespace(), mt.Version())))
-		return serrs
+	if cdt == nil {
+		// initialize config tree if nil is provided
+		cdt = cdata.NewTree()
 	}
 
-	m.config = cd
+	for _, m := range mts {
+		config := cdt.Get(m.Namespace().Strings())
 
-	typ, serr := core.ToPluginType(m.Plugin.TypeName())
-	if serr != nil {
-		return []serror.SnapError{serror.New(err)}
-	}
+		if config == nil {
+			config = cdata.NewNode()
+		}
+		m.config = config
 
-	// merge global plugin config
-	if m.config != nil {
-		m.config.ReverseMerge(p.Config.Plugins.getPluginConfigDataNode(typ, m.Plugin.Name(), m.Plugin.Version()))
-	} else {
-		m.config = p.Config.Plugins.getPluginConfigDataNode(typ, m.Plugin.Name(), m.Plugin.Version())
-	}
+		typ, serr := core.ToPluginType(m.Plugin.TypeName())
+		if serr != nil {
+			return []serror.SnapError{serror.New(err)}
+		}
 
-	// When a metric is added to the MetricCatalog, the policy of rules defined by the plugin is added to the metric's policy.
-	// If no rules are defined for a metric, we set the metric's policy to an empty ConfigPolicyNode.
-	// Checking m.policy for nil will not work, we need to check if rules are nil.
-	if m.policy.HasRules() {
-		if m.Config() == nil {
-			serrs = append(serrs, serror.New(fmt.Errorf("Policy defined for metric, (%s) version (%d), but no config defined in manifest", mt.Namespace(), mt.Version())))
-			return serrs
+		// merge global plugin config
+		if m.config != nil {
```

Comment:
By the way, this diff between master and this PR changes is very difficult to read even there is not much change in the code - just  fragment of code related with config has been put in `for` loop over metrics + removing the fragment as below: 

```
if m.Config() == nil {
serrs = append(serrs, serror.New(fmt.Errorf("Policy defined for metric, (%s) version (%d), but no config defined in manifest", mt.Namespace(), mt.Version())))
return serrs
```
because it has never happen - it was unreachable code. But as You caught out, the checking if `m.config !=nil` is exactly the same situation and does no sense any more. So, I will remove that too. Thanks